kingpin/actors/aws/base.py

```diff
@@ -86,19 +90,23 @@ def __init__(self, *args, **kwargs):
         # By default, we will try to let Boto handle discovering its
         # credentials at instantiation time. This _can_ result in synchronous
         # API calls to the Metadata service, but those should be fast.
-        key = None
-        secret = None
-
+        #
         # In the event though that someone has explicitly set the AWS access
         # keys in the environment (either for the purposes of a unit test, or
         # because they wanted to), we use those values.
-        if aws_settings.AWS_ACCESS_KEY_ID and aws_settings.AWS_SECRET_ACCESS_KEY:
-            key = aws_settings.AWS_ACCESS_KEY_ID
-            secret = aws_settings.AWS_SECRET_ACCESS_KEY
+        #
+        # Note: these get defualted to None in aws_settings if they are not
+        # found which will tell boto3 to fallback to default behavior.
+        boto3_client_kwargs = {}
+        boto3_client_kwargs["aws_access_key_id"] = aws_settings.AWS_ACCESS_KEY_ID
+        boto3_client_kwargs[
+            "aws_secret_access_key"
+        ] = aws_settings.AWS_SECRET_ACCESS_KEY
+        boto3_client_kwargs["aws_session_token"] = aws_settings.AWS_SESSION_TOKEN
 
         # Establish connection objects that don't require a region
         self.iam_conn = boto3.client(
-            "iam", aws_access_key_id=key, aws_secret_access_key=secret
+            service_name="iam", config=None, **boto3_client_kwargs
         )
```

Comment:
1. I had assumed the use of the kwargs was to ensure we don’t pass none values. What is the motivation of using a scattered kwargs dictionary here? 
2. What is the motivation for the change from positional to kwargs and explicitly setting config=None? Is that critical? It looks to only be done in this case.

Reply:
So for no. 1, we are actually doing the opposite. We want to be able to easily pass `None` values. For no. 2, I am moving away from positional args here just to make it safer than mixing positional/named/kwargs. (And yes, to specifically to set `config=None` so that it just matches the calls below.

Reply:
You bring up a good point in number 1 tho. Let me quickly test if passing a dict with None values actually passes that None to the function or not.